src/protocol/gossip/impl/connectivity.cpp

```diff
@@ -216,44 +204,27 @@ namespace libp2p::protocol::gossip {
 
     if (!ctx->outbound_stream) {
       // make stream for writing
-      dial(ctx, true);
+      dialOverExistingConnection(ctx);
     } else {
       flush(ctx);
     }
   }
 
-  void Connectivity::dial(const PeerContextPtr &ctx,
-                          bool connection_must_exist) {
-    using C = Host::Connectedness;
-
+  void Connectivity::dial(const PeerContextPtr &ctx) {
     if (ctx->is_connecting || ctx->outbound_stream) {
       return;
     }
 
-    if (ctx->banned_until != 0 && connection_must_exist) {
-      // unban outbound connection only if inbound one exists
-      unban(ctx);
-    } else {
-      connectable_peers_.erase(ctx->peer_id);
-    }
+    connectable_peers_.erase(ctx->peer_id);
 
     peer::PeerInfo pi = host_->getPeerRepository().getPeerInfo(ctx->peer_id);
-    if (ctx->dial_to) {
-      pi.addresses = {ctx->dial_to.value()};
-    }
```

Comment:
Hmm, cannot understand why this code is gone.

Reply:
this field is gone, address repo is used now